sql/core/src/test/scala/org/apache/spark/sql/expressions/ExpressionInfoSuite.scala

```diff
@@ -105,11 +105,34 @@ class ExpressionInfoSuite extends SparkFunSuite with SharedSparkSession {
     }
   }
 
+  test("SPARK-32870: Default expressions in FunctionRegistry should have their " +
+    "usage, examples and since filled") {
+    val ignoreSet = Set(
+      "org.apache.spark.sql.catalyst.expressions.TimeWindow")
+
+    spark.sessionState.functionRegistry.listFunction().foreach { funcId =>
+      val info = spark.sessionState.catalog.lookupFunctionInfo(funcId)
+      if (!ignoreSet.contains(info.getClassName)) {
+        withClue(s"Function '${info.getName}', Expression class '${info.getClassName}'") {
+          assert(info.getUsage.nonEmpty)
+          assert(info.getExamples.startsWith("\n    Examples:\n"))
+          assert(info.getExamples.endsWith("\n  "))
+          assert(info.getSince.matches("[0-9]+\\.[0-9]+\\.[0-9]+"))
```

Comment:
Looks these checks have already been done in the `ExpressionInfo` constructor?

Reply:
These ones are a bit more strict. 
For example this extra space made the Example for `typeof` not appear in http://spark.apache.org/docs/latest/api/sql/#typeof

https://github.com/apache/spark/pull/29743/files#diff-25282ab1377a3d87999d1d0d7a8ec270R208-R214

I didn't want to add these checks to the constructor, because I'm afraid that they might break some UDFs and also I have no way to exclude some from the checks.

Reply:
> I didn't want to add these checks to the constructor, because I'm afraid that they might break some UDFs and also I have no way to exclude some from the checks.

If we make the checks more strict in the ctor and we find existing expressions throwing exceptions, I think it is okay just to fix them. Any problem there?

Reply:
No, problems with current ones - only some dummy functions in tests.